x/auth/ante/basic_test.go

```diff
@@ -100,14 +101,21 @@ func (suite *AnteTestSuite) TestConsumeGasForTxSize() {
 	msg := testdata.NewTestMsg(addr1)
 	feeAmount := testdata.NewTestFeeAmount()
 	gasLimit := testdata.NewTestGasLimit()
-	suite.txBuilder.SetMsgs(msg)
+	suite.Require().NoError(suite.txBuilder.SetMsgs(msg))
 	suite.txBuilder.SetFeeAmount(feeAmount)
 	suite.txBuilder.SetGasLimit(gasLimit)
 	suite.txBuilder.SetMemo(strings.Repeat("01234567890", 10))
 
 	privs, accNums, accSeqs := []crypto.PrivKey{priv1}, []uint64{0}, []uint64{0}
-	tx := suite.CreateTestTx(privs, accNums, accSeqs, suite.ctx.ChainID())
-	txBytes, err := json.Marshal(tx)
+	tx, err := suite.CreateTestTx(privs, accNums, accSeqs, suite.ctx.ChainID())
+	suite.Require().NoError(err)
+
+	var txBytes []byte
+	if suite.clientCtx.TxConfig.SignModeHandler().DefaultMode() == signing.SignMode_SIGN_MODE_DIRECT {
+		txBytes, err = suite.clientCtx.TxConfig.TxJSONEncoder()(tx)
+	} else {
+		txBytes, err = json.Marshal(tx)
+	}
```

Comment:
In amino legacy mode, when I do `suite.clientCtx.TxConfig.TxJSONEncoder()(tx)`, I get the following message:
```
Expected nil, but got: &errors.errorString{s:"Unregistered interface types.Msg"}
```

That's why I'm using json.Marshal here (like before). However, i guess the correct way would be to use `TxConfig.TxJSONEncoder()`. How can I register [TestMsg](https://github.com/cosmos/cosmos-sdk/blob/5c0e3b4de5b808765b1ec4793cfdba30401e023e/testutil/testdata/proto.pb.go#L542) in Amino's registry?